### PR TITLE
Add docstring for `generate_unit_locations` 

### PR DESCRIPTION
**Pull request by h-mayorquin**
This function is missing a docstring.

### CODE REVIEW

File: src/spikeinterface/core/generate.py [hunk 1]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+    minimum_z : float, default 5.0
+        The minimum z-coordinate value for generated unit locations.
+    maximum_z : float, default 40.0
+        The maximum z-coordinate value for generated unit locations.
+    minimum_distance : float, default 20.0
+        The minimum allowable distance in micrometers between any two units
+    max_iteration : int, default 100
+        The maximum number of iterations to attempt generating unit locations that meet
+        the minimum distance constraint (default is 100).
+    distance_strict : bool, optionaldefault False
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2031]:
```suggestion
    distance_strict : bool, default: False
```

File: src/spikeinterface/core/generate.py [hunk 2]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+    the function can either raise an exception or issue a warning based on the `distance_strict` flag.
+
+    Parameters
+    ----------
+    num_units : int
+        Number of units to generate locations for.
+    channel_locations : numpy.ndarray
+        A 2D array of shape (num_channels, 2), where each row represents the (x, y) coordinates
+        of a channel.
+    margin_um : float, default 20.0
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2019]:
```suggestion
    margin_um : float, default: 20.0
```

File: src/spikeinterface/core/generate.py [hunk 3]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        The margin to add around the minimum and maximum x and y channel coordinates when
+        generating unit locations
+    minimum_z : float, default 5.0
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2022]:
```suggestion
    minimum_z : float, default: 5.0
```

File: src/spikeinterface/core/generate.py [hunk 4]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        The minimum z-coordinate value for generated unit locations.
+    maximum_z : float, default 40.0
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2024]:
```suggestion
    maximum_z : float, default: 40.0
```

File: src/spikeinterface/core/generate.py [hunk 5]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        The maximum z-coordinate value for generated unit locations.
+    minimum_distance : float, default 20.0
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2026]:
```suggestion
    minimum_distance : float, default: 20.0
```

File: src/spikeinterface/core/generate.py [hunk 6]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        The minimum allowable distance in micrometers between any two units
+    max_iteration : int, default 100
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2028]:
```suggestion
    max_iteration : int, default: 100
```

File: src/spikeinterface/core/generate.py [hunk 7]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        The maximum number of iterations to attempt generating unit locations that meet
+        the minimum distance constraint (default is 100).
+    distance_strict : bool, optionaldefault False
+        If True, the function will raise an exception if a solution meeting the distance
+        constraint cannot be found within the maximum number of iterations. If False, a warning
+        will be issued (default is False).
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2034]:
```suggestion
        will be issued.
```
we already have the default unless you want to say it twice.

File: src/spikeinterface/core/generate.py [hunk 8]
```diff
@@ -1994,6 +1994,53 @@ def generate_unit_locations(
...
+        generating unit locations
+    minimum_z : float, default 5.0
+        The minimum z-coordinate value for generated unit locations.
+    maximum_z : float, default 40.0
+        The maximum z-coordinate value for generated unit locations.
+    minimum_distance : float, default 20.0
+        The minimum allowable distance in micrometers between any two units
+    max_iteration : int, default 100
+        The maximum number of iterations to attempt generating unit locations that meet
+        the minimum distance constraint (default is 100).
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:2030]:
```suggestion
        the minimum distance constraint.
```
same

**Review by zm711 — COMMENTED** [review]
This looks good to me. Just a couple style fixes.

**Review by alejoe91 — APPROVED** [review]

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
merci.

**Pull request merged**